imagegw/shifter_imagegw/transfer.py

```diff
@@ -219,24 +317,56 @@ def check_file(filename, system, logger=None):
         basepath = system['ssh']['imageDir']
     image_fn = os.path.split(filename)[1]
     target_fn = os.path.join(basepath, image_fn)
-    ls_cmd = sh_cmd(system, 'ls', target_fn)
+    if import_image:
+        ls_cmd = sh_cmd(system, 'ls', filename)
+    else:
+        ls_cmd = sh_cmd(system, 'ls', target_fn)
     ret = _exec_and_log(ls_cmd, logger)
 
     if ret == 0:
         return True
     return False
 
 
-def transfer(system, image_path, metadata_path=None, logger=None):
+def hash_file(filename, system, logger=None):
+    """
+    Calculate a hash of the image file,
+    because this can be remote or local, 
+    need to use a separate helper executable fasthash.py
+    assume it is in the path already
+    """
+    if system['accesstype'] == 'local':
+        sh_cmd = _sh_cmd
```

Comment:
Could potentially just run it in place here (e.g. import fasthash and call it).  But this should be okay too.

Reply:
I don' think this would work for the remote case that does "ssh fasthash.py image". It'll still need to be in the path on the remote system.